renderdoc/driver/d3d12/d3d12_command_list_wrap.cpp

```diff
@@ -812,11 +812,33 @@ void WrappedID3D12GraphicsCommandList::ResourceBarrier(UINT NumBarriers,
   for(UINT i = 0; i < NumBarriers; i++)
   {
     barriers[i] = pBarriers[i];
-    barriers[i].Transition.pResource = Unwrap(barriers[i].Transition.pResource);
+    if(barriers[i].Type == D3D12_RESOURCE_BARRIER_TYPE_TRANSITION)
+    {
+      barriers[i].Transition.pResource = Unwrap(barriers[i].Transition.pResource);
 
-    // hack while not all resources are wrapped
-    if(barriers[i].Transition.pResource == NULL)
-      barriers[i].Transition.pResource = pBarriers[i].Transition.pResource;
+      // hack while not all resources are wrapped
+      if(barriers[i].Transition.pResource == NULL)
```

Comment:
This hack can be deleted from all paths now, this was only while I was doing early testing and hadn't wrapped some simple things yet. There are still some unwrapped resources but they're not needed for a 'hello world' type application, so it's OK to fail here - they just need to be implemented.